### PR TITLE
Fix Quantity onchange on source for Framework Agreement

### PR DESCRIPTION
**Pull request by yvaucher**

### CODE REVIEW

File: framework_agreement_sourcing/model/logistic_requisition_source.py [hunk 1]
```diff
@@ -357,8 +357,8 @@ def onchange_quantity(self):
                 or not self.proposed_product_id):
             return
         agreement = self._get_best_agreement()
-        self.framework_agreement_id = agreement.id
         if agreement:
+            self.framework_agreement_id = agreement.id
             return self._check_enought_qty(agreement)
```
Comment (gurneyalex) [inline framework_agreement_sourcing/model/logistic_requisition_source.py:362]: you need a return statement in the else branch to as xmlrpc methods are not allowed to return None

Comment (lepistone) [inline framework_agreement_sourcing/model/logistic_requisition_source.py:362]: This is a new-style onchange, and if I understand correctly, it changes attributes of `self` like @yvaucher does here. Same thing for new-style computed fields. This even works when the record is not saved into the database yet.

On the other hand, here a warning is returned. That probably works, but it is not clear from the doc.

**Comment by lepistone** [comment]
:+1: , thanks

File: framework_agreement_sourcing/model/logistic_requisition_source.py (hunk 1 above)
Comment (guewen) [inline framework_agreement_sourcing/model/logistic_requisition_source.py:362]: Yes, returning a warning works. New onchange methods do not need to explicitly returns a value different that `None` when they do not return a warning, I guess that the framework deals with that itself.

Comment (lepistone) [inline framework_agreement_sourcing/model/logistic_requisition_source.py:362]: thanks. I didn't find that in the doc, so I opened https://github.com/odoo/odoo/issues/3968

**Comment by guewen** [comment]
:+1:

**Comment by jgrandguillaume** [comment]
:+1:

**Pull request merged**